### ISSUE TITLE
SignInActivity is not finished after successfully signin.

### ISSUE THREAD

**Issue by sandro-csimas** [vinli/android-net#51]
Please take a look at #50

**Comment by sandro-csimas** [comment]
As SignInActivity is opened when calls Vinli.signIn, the activity should also be closed by the library after a successfully signin. Here I'm having problems because I can signin with success, but the SignInActivity stays open.
First I was trying to execute Vinli.signIn with a pending intent created by PendingIntent.getBroadcast. My goal was to keep the same activity opened and receive a intent after Vinli signin.
````
 @OnClick(R.id.add_vinli_vehicle)
 void onAddVinliVehicleClick() {
   PendingIntent pendingIntent = PendingIntent.getBroadcast(this, 0, new Intent(ACTION_VINLI_SIGNIN), 0);
   Vinli.signIn(this, getString(R.string.vinli_client_id), getString(R.string.vinli_redirect_uri), pendingIntent);
}
````
````
broadcastReceiver = new BroadcastReceiver() {
  @Override
  public void onReceive(Context context, Intent intent) {
    switch(intent.getAction()) {
      case ACTION_VINLI_SIGNIN:
        listVinliVehicles(intent);
        break;
    }
  }
};
IntentFilter intentFilter = new IntentFilter(ACTION_VINLI_SIGNIN);
registerReceiver(broadcastReceiver, intentFilter);
````
The code above works fine and receives the broadcast with ACTION_VINLI_SIGNIN, but the SignInActivity stays opened.

After that, with no succes, i tried to follow the example you provided executing Vinli.signIn with a pending intent created with PendingIntent.getActivity.
````
@OnClick(R.id.add_vinli_vehicle)
void onAddVinliVehicleClick() {
  PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, new Intent(this, VinliSigninSuccessActivity.class), 0);
  Vinli.signIn(this, getString(R.string.vinli_client_id), getString(R.string.vinli_redirect_uri), pendingIntent);
}
````
````
public class VinliSigninSuccessActivity extends ClosicActivity {
  public static final String ACTION_VINLI_SIGNIN = "com.closic.intent.action.VINLI_SIGNIN";
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    sendBroadcast(new Intent(ACTION_VINLI_SIGNIN));
    finish();
  }
}
````
The code above also is not working and the SignInActivity stays opened.
I think you can improve your library including a finish call after signin.

**Comment by cmc5788** [comment]
None of our own apps our others have reported this particular issue, and all use the net SDK for sign in. We'll need need to reproduce it internally before modifying the net SDK surrounding sign in, but we'll take a look.

**Comment by dofchinick** [comment]
Moved to vinli/platform#666

**Comment by cmc5788** [comment]
@joshberidon - at any point when SignInActivity finishes and activity.

https://github.com/vinli/android-net/blob/028a3ab4c4c73f9d5771c0ba6908d2ec1e0f70de/android-net/src/main/java/li/vin/net/SignInActivity.java#L121

Need to test this against Beagle, L&K, Drive and ensure that it doesn't break 1st pty app signin flow.

**Comment by cmc5788** [comment]
Closed by #69
